go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -889,8 +889,13 @@ func (h *defaultNetworkControllerEventHandler) UpdateResource(oldObj, newObj int
 			return h.oc.addUpdateLocalNodeEvent(newNode, nodeSyncsParam)
 		} else {
 			_, syncZoneIC := h.oc.syncZoneICFailed.Load(newNode.Name)
-			// Check if the node moved from local zone to remote zone and if so syncZoneIC should be set to true
-			syncZoneIC = syncZoneIC || h.oc.isLocalZoneNode(oldNode)
+			// Check if the node moved from local zone to remote zone and if so syncZoneIC should be set to true.
+			// Also check if node subnet changed, so static routes are properly set
+			syncZoneIC = syncZoneIC || h.oc.isLocalZoneNode(oldNode) || nodeSubnetChanged(oldNode, newNode)
+			if syncZoneIC {
+				klog.Infof("Node %s in remote zone %s needs interconnect zone sync up.",
```

Comment:
can you double check to make sure this log is not spitted out too often, causing spam? I anticipate it won't be given the syncZoneIC will be true under special conditions only, but better safe than sorry

Reply:
ack; will do.

Reply:
@tssurya I ran a few e2e tests after converting a ind cluster to dualstack and made sure no chatty logs happend.
```
❯ kn logs $POD -c ovnkube-node | grep -i 'in remote zone' | wc -l                                                                         │        ../test/e2e/.vscode/
0
```

Reply:
mark as resolved.